pages/index.js

```diff
@@ -1,14 +1,49 @@
 import ActivityCard from "@/components/Activitylist/ActivityCard";
 import ActivityList from "@/components/Activitylist/ActivityList";
+import useSWR from "swr";
+import { useState } from "react";
 
+export default function HomePage() {
+  const { data: activities, isLoading, error } = useSWR("/api/activities");
+  const [search, setSearch] = useState("");
 
+  if (isLoading) return <p>Loading...</p>;
+  if (error) return <p>Error to load the Page.</p>;
+
+  const filterActivities = activities.filter((activity) => {
+    const match = search.toLowerCase();
+
+    const title = activity.title?.toLowerCase() || "";
+    const category = activity.category?.toLowerCase() || "";
+    const country = activity.country?.toLowerCase() || "";
+
+    return (
+      title.includes(match) ||
+      category.includes(match) ||
+      country.includes(match)
```

Comment:
little trick if you only use strings: 
```suggestion
      `${title}|${category}|${country}`.includes(match)
```
The `|` between the elements are only there to separate the elements, you can use any type of unusual character like "_" "%" or "/"